File: frontend/src/types/archive.ts
// types/archive.ts

/** 모아보기에 쌓이는 첨부 종류. 동영상은 받지 않는다(용량이 커서 제외) */
export type AttachmentKind = 'file' | 'image';

/** 채팅에서 주고받은 첨부 하나 */
export interface Attachment {
  id: string;
  kind: AttachmentKind;
  name: string;
  /** 파일 크기(byte) */
  sizeBytes: number;
  /** 올린 시각 (ISO 8601). 연-월로 묶는 기준 */
  uploadedAt: string;
  /** 다운로드·미리보기 URL (사진은 썸네일) */
  url?: string;
}

/** 진도 로드맵의 한 지점 */
export interface ProgressMilestone {
  id: string;
  /** 예: '7월까지 20강 완료' */
  title: string;
  /** 완료 / 진행 중 / 예정 */
  status: 'done' | 'current' | 'upcoming';
}

/** 목표 하나의 진도 로드맵 */
export interface GoalProgress {
  goalId: string;
  /** 목표 이름 (Goal.title) */
  goalTitle: string;
  /** 목표를 시작한 날 (ISO 8601) */
  startedAt: string;
  /** 목표를 끝낸 날. 아직 진행 중이면 없다 */
  completedAt?: string;
  milestones: ProgressMilestone[];

  /** 이 목표에 쓴 집중 시간(초). 집중 세션에 목표 id가 붙어야 계산할 수 있다 */
  focusedSeconds?: number;
  /** 이 목표에서 완료한 투두 개수 */
  completedTodoCount?: number;
  /** 가장 많이 집중한 달 (YYYY-MM) */
  bestMonth?: string;
}

File: backend/app/schemas/archive.py
"""모아보기(진도) 스키마.

프론트 types/archive.ts GoalProgress 와 1:1 (camelCase).
"""

from app.schemas.common import CamelModel, UtcDatetime


class AttachmentOut(CamelModel):
    """모아보기 첨부 (types/archive.ts Attachment).

    url 은 항상 원본이다(뷰어·다운로드용). 목록에 작게 그릴 때 쓰라고
    사진(image)은 thumbnail_url 을 함께 준다(api.md §3.6, §8-3-2).
    """

    id: str
    kind: str  # file | image
    name: str
    size_bytes: int
    uploaded_at: UtcDatetime
    url: str | None = None
    # 사진만 있다. 없으면 프론트가 url(원본)로 대체한다.
    thumbnail_url: str | None = None


class ProgressMilestoneOut(CamelModel):
    id: str
    title: str
    status: str  # done | current | upcoming


class GoalProgressOut(CamelModel):
    """목표 진도 로드맵. 진행률(%)은 프론트가 done÷전체로 계산한다."""

    goal_id: str
    goal_title: str
    started_at: UtcDatetime
    completed_at: UtcDatetime | None = None
    milestones: list[ProgressMilestoneOut]

    # 집계 3종. 없으면 프론트가 해당 칸을 그리지 않는다(초기엔 생략 가능).
    focused_seconds: int | None = None  # 집중 세션에 goal_id 필요(4단계)
    completed_todo_count: int | None = None
    best_month: str | None = None  # 완료한 할 일이 가장 많았던 달 'YYYY-MM'
